### PR TITLE
[BUGFIX] Fix use of column.partition metric in HistogramSingleBatchParameterBuilder to more accurately handle errors

### PR DESCRIPTION
**Pull request by alexsherstinsky**
### Scope
* When `HistogramSingleBatchParameterBuilder` uses `column.partition` metric, only `None` `bins` should generate an error.  If, however, each `bin` is `np.nan`, that is not an error; rather, it is the accurate reflection of a column's data values.
* Updated existing test and added a new test to reflect the more accurate behavior.

Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

Changes proposed in this pull request:
- JIRA: GREAT-761/GREAT-1247
-
-

After submitting your PR, CI checks will run and @cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: tests/rule_based_profiler/parameter_builder/test_histogram_single_batch_parameter_builder.py [hunk 1]
```diff
@@ -136,6 +136,59 @@ def test_histogram_single_batch_parameter_builder_alice_null_bins(
...
+    }
+
+    parameter_builder: ParameterBuilder = HistogramSingleBatchParameterBuilder(
+        name="my_name",
+        bins="auto",
+        evaluation_parameter_builder_configs=None,
+        data_context=data_context,
+    )
+
+    metric_domain_kwargs: dict = {"column": "my_column"}
```
Comment (NathanFarmer) [inline tests/rule_based_profiler/parameter_builder/test_histogram_single_batch_parameter_builder.py:178]: Is this a real column or a dummy? Why does this produce `NaN` bins?

File: great_expectations/rule_based_profiler/parameter_builder/histogram_single_batch_parameter_builder.py [hunk 2]
```diff
@@ -130,8 +129,7 @@ def _build_parameters(
             FULLY_QUALIFIED_PARAMETER_NAME_VALUE_KEY
         ]
 
-        element: Any
-        if bins is None or all(pd.isnull(element) for element in bins):
+        if bins is None:
```
Comment (NathanFarmer) [inline great_expectations/rule_based_profiler/parameter_builder/histogram_single_batch_parameter_builder.py:132]: ❤️

**Review by NathanFarmer — APPROVED** [review]
One minor question about the test!

File: tests/rule_based_profiler/parameter_builder/test_histogram_single_batch_parameter_builder.py (hunk 1 above)
Comment (alexsherstinsky) [reply tests/rule_based_profiler/parameter_builder/test_histogram_single_batch_parameter_builder.py:178]: @NathanFarmer I mocked it to produce `NaN` values -- as in the below.

**Pull request merged**